dlib/cmake_utils/find_blas.cmake

```diff
@@ -19,8 +19,11 @@
 #  openmp_libraries         - Set to Intel's OpenMP library if and only if we
 #                             find the MKL.
 
+INCLUDE(CheckFunctionExists)
+
 # setting this makes CMake allow normal looking if else statements
 SET(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
+SET(CMAKE_REQUIRED_LIBRARIES "cblas")
```

Comment:
This line should be immediately before the use of `CHECK_FUNCTION_EXISTS` since it's the input to `CHECK_FUNCTION_EXISTS`.